app/lib/common/widgets/share/action/share_space_object_action.dart

```diff
@@ -167,4 +219,25 @@ class ShareSpaceObjectActionUI extends StatelessWidget {
       _ => null,
     };
   }
+
+  Future<RefDetails?> getRefDetails({
+    required WidgetRef ref,
+    required SpaceObjectDetails objectDetails,
+  }) async {
+    final objectId = objectDetails.objectId;
+    switch (objectDetails.objectType) {
```

Comment:
No, this require here.
This is for getting `RedDetails` of source object whereas `onTapPin` is for destination object.

`objectDetails.objectType == ObjectType.calendarEvent` mean you are in Calendar Details screen and this code will get you `RefDetails` of calendar object.

Whereas `onTapPin` define destination Pin Object in which you want to put Calendar source object.

I hope this will clarify the above code. Still happy to elaborate more if needed.

Reply:
I understand what the API does / how it works _right_ now. I am arguing that this should be changed because it is wrong and leads to unnecessarily complicated code like this.

Simply put. Why is `onTapPin: Function(ObjectDetails { ObjectType.pin, ActerPin })` and not simply `onTapPin : Function(ActerPin)`? This seems like an unnecessary attempt of generalizing something that is _always_ specific...

And that more specific API would then allow to simplify this code to : 
```
onTapPin: (pin) async => await attachRefDetailToPin(
            context: context,
            ref: ref,
            refDetails: await pin.refDetails())
```

I'd rather do these smaller refactors that show an earlier approach can be improved massively sooner rather than later. As with every use the API gets more entrenched in the way it is even if that means more complicated code around it. and changing it later becomes harder.

If you still don't understand what I mean, let me know and I push a commit refactoring it.